### PR TITLE
Parse elf file to find out string table

### PR DESCRIPTION
**Pull request by howard0su**
Parse .o file is not the best approach as some string can get optmized
out in link stage. LTO .o file even doesn't contain a string table.

### CODE REVIEW

File: src/target/common/devo/devo.ld [hunk 1]
```diff
@@ -34,13 +34,14 @@ SECTIONS
...
 		*(.text*)	/* Program code */
-		. = ALIGN(4);
-		*(.rodata*)	/* Read-only data */
-		. = ALIGN(4);
 	} >rom
 
+    .rodata :{
+		. = ALIGN(4);
+		*(.rodata*)	/* Read-only data */
+    }
```
Comment (hexfet) [inline src/target/common/devo/devo.ld]: Whitespace issue here and above.

**Comment by howard0su** [comment]
please review.

File: src/target/common/devo/devo.ld [hunk 2]
```diff
@@ -34,13 +34,14 @@ SECTIONS
 		*(.vectors)	/* Vector table */
 		INCLUDE optimize.ld
 		. = _crc_offset;
-                KEEP(*(.crc))
+        KEEP(*(.crc))
 		*(.text*)	/* Program code */
```
[the remark this answers is not in the document]
Comment (howard0su) [reply src/target/common/devo/devo.ld]: already fixed in the latest rev.

File: src/target/common/devo/devo.ld [hunk 3]
```diff
@@ -34,13 +34,14 @@ SECTIONS
...
-		. = ALIGN(4);
-		*(.rodata*)	/* Read-only data */
-		. = ALIGN(4);
 	} >rom
 
+    .rodata :{
+		. = ALIGN(4);
+		*(.rodata*)	/* Read-only data */
+    }
 	/* C++ Static constructors/destructors, also used for __attribute__
```
[the remark this answers is not in the document]
Comment (howard0su) [reply src/target/common/devo/devo.ld]: better to be safe to add it. I believe some well known section can omit it. Last revision I missed >rom but still works fine.

**Pull request merged**